indigo_social/models.py

```diff
@@ -71,17 +49,15 @@ class UserProfile(models.Model):
     twitter_username = models.CharField(max_length=256, blank=True, null=True, default='')
     linkedin_profile = models.URLField(blank=True, null=True, default='')
 
-    def save(self):
-        # TODO: Generate and save the nonce here
-        self.profile_photo = resize_photo(self.profile_photo, self.user)
-
-        super(UserProfile, self).save()
+    def generate_nonce(self):
+        self.profile_photo_nonce = ''.join(random.sample(string.ascii_lowercase, 8))
+        self.save()
```

Comment:
A good rule of thumb is not to call `save()` from inside a model. In some cases you may not want to save it, or it's being saved prematurely, etc. Instead, the part of the code closest to the user's request (eg. the view) should call `save()` once it knows everything is ready.